File: src/main/java/com/epam/eng/pattern/bridge/GameConsole.java
package com.epam.eng.pattern.bridge;

public class GameConsole {
    private Joystick joystick = new Joystick();
    private Gamepad gamepad = new Gamepad();
    private String currentController;

    public GameConsole() {
        currentController = "Joystick";
    }

    public void setController(String controller) {
        if (controller.equals("Joystick")) {
            currentController = "Joystick";
        } else if (controller.equals("Gamepad")) {
            currentController = "Gamepad";
        } else {
            throw new IllegalArgumentException("Invalid controller");
        }
    }

    public void moveUp() {
        if (currentController.equals("Joystick")) {
            joystick.moveUp();
        } else if (currentController.equals("Gamepad")) {
            gamepad.pressButton("Up");
        }
    }
}

class Joystick {
    public void moveUp() {
        // move joystick up
    }
}

class Gamepad {
    public void pressButton(String button) {
        // press gamepad button
    }
}

File: ts/bridge/GameConsole.ts
class GameConsole {
    private joystick: Joystick = new Joystick();
    private gamepad: GamePad = new GamePad();
    private currentController: string;

    constructor() {
        this.currentController = "Joystick";
    }

    public setController(controller: string): void {
        if (controller === "Joystick") {
            this.currentController = "Joystick";
        } else if (controller === "Gamepad") {
            this.currentController = "Gamepad";
        } else {
            throw new Error("Invalid controller");
        }
    }

    public moveUp(): void {
        if (this.currentController === "Joystick") {
            this.joystick.moveUp();
        } else if (this.currentController === "Gamepad") {
            this.gamepad.pressButton("Up");
        }
    }
}

class Joystick {
    public moveUp(): void {
        // move joystick up
    }
}

class GamePad {
    public pressButton(button: string): void {
        // press gamepad button
    }
}

var gameConsole = new GameConsole();
gameConsole.moveUp();

// todo: consider abstract GameController, so that Joystick and Gamepad can extend them.
